### PR TITLE
MEP-19 add guess word

### PR DESCRIPTION
**Pull request by qiming5**

### CODE REVIEW

File: app/controllers/api/v1/admin/guess_word_dictionaries_controller.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    GuessWordDictionary.transaction do
+      uploaded_file = params.permit(:file)[:file]
+      filename = uploaded_file.tempfile.path
+
+      File.foreach(filename) do |word|
+        word = word.strip
+        if word.length.positive?
+          read += 1
+
+          entry = GuessWordDictionary.create(word: word)
```
Comment (ang202) [inline app/controllers/api/v1/admin/guess_word_dictionaries_controller.rb:54]: @qiming5  Is it better if we set it in an array and use insert_all to store the words? Because the dictionary will have approx 100,000 words so it could cause performance issue

**Comment by ang202** [comment]
@qiming5  Can we include current_guess_word_submission in guess_word list? So I can direct get the id for guess_word_submission

**Review by ang202 — APPROVED** [review]

**Comment by qiming5** [comment]
@ang202 please check if this is okay, I have decided to load the guess word submissions and manually associate them on the controllers

the alternative is to only show guess_word entries that has a user's guess_word_submission
because i cant do `GuessWord.includes(:guess_word_submissions)` as this would show submission by everyone
i could filter it out inside serializer with `object.guess_word_submissions.find { |user| user.id == current_user.id}`, but that would be inefficient at a larger scale
doing `object.guess_word_submissions.where(user: current_user)` inside the serializer would result in a N+1 query which is what stumped me and stopped me from adding it previously

if i do  `GuessWord.includes(:guess_word_submissions).where(guess_word_submissions: {user: current_user})` guess_word with no user submission are omitted

There is a way to do custom join conditions with arel, but rails wont preload them so associating will have to be done manually
https://stackoverflow.com/a/77095474 https://stackoverflow.com/a/42044265

**TLDR:**
with everything else considered, i dont see a way to solve this problem
therefore my approach is to query for submissions and manually associating it inside the controller, while allowing all guess_word to be shown in one request

if that's not necessary i can tidy it up to only show guess_word with guess_word_submission
if i overlooked a better method to solve this problem, please let me know

Sample response with `?with_submission=true`:
```json
{
  "guess_words": [
    {
      "id": "d13660b0-5630-45a5-822c-7bfc92d74755",
      "subject_id": "59310c47-a8b0-4595-913a-2f4873ea85a3",
      "description": "In minus rem. Sed asperiores quod. Ut eos suscipit.",
      "attempts": 7,
      "reward_points": 5,
      "answer_length": 8,
      "start_at": "2024-08-01T17:33:31.898+08:00",
      "end_at": "2024-08-09T10:55:58.135+08:00",
      "created_at": "2024-08-05T18:37:13.550+08:00",
      "updated_at": "2024-08-05T18:37:13.550+08:00",
      "user_guess_word_submissions": [
        {
          "id": "5f08e252-a26d-4285-b0f0-f9890e0766ec",
          "guess_word_id": "d13660b0-5630-45a5-822c-7bfc92d74755",
          "user_id": "8dbad404-a036-4006-8ec0-0b538e3d2c04",
          "status": "success",
          "guesses_count": 0,
          "completed_at": "2024-08-05T18:37:13.557+08:00",
          "created_at": "2024-08-05T18:37:13.562+08:00",
          "updated_at": "2024-08-05T18:37:13.562+08:00",
          "guess_word_answer": "incidunt",
          "guesses": []
        }
      ],
      "subject": {
        "id": "59310c47-a8b0-4595-913a-2f4873ea85a3",
        "name": "et-4",
        "code": "YPA2R",
        "curriculum_id": "032d6d04-6269-4a1d-b684-ecb016a16b09"
      }
    },
```

**Comment by qiming5** [comment]
@ang202 thx for waiting, should be good now
